R/bm-write-csv.R

```diff
@@ -10,25 +10,39 @@ write_csv <- Benchmark(
   "write_csv",
   setup = function(source = names(known_sources),
                    writer = "arrow",
+                   compression = "uncompressed",
                    input = c("arrow_table", "data_frame")) {
     writer <- match.arg(writer, c("arrow", "data.table", "vroom", "readr", "base"))
+    compression <- match.arg(compression, c("gzip", "uncompressed"))
     input <- match.arg(input)
 
     # source defaults are retrieved from the function definition (all available
     # known_sources) and then read the source in as a data.frame
     source <- ensure_source(source)
     df <- read_source(source, as_data_frame = match.arg(input) == "data_frame")
 
+    if (compression != "uncompressed") {
+      if (compression == "gzip") {
+        ext <- ".csv.gz"
+      } else {
+        ext <- paste0(".csv.", compression)
+      }
+    } else {
+      ext <- ".csv"
+    }
```

Comment:
Yeah I had that thought too. One question though that impacts this:
> do we want to benchmark all the arrow compression options?

Any thoughts on this?

Reply:
In looking through the sources code a lot this sprint, it mostly assumes CSVs can be uncompressed or gzipped. That's irrelevant for this since it's writing here, but not a terrible take generally. But if you want to add in all the rest for kicks, go for it. I'd make at least those two defaults so they get run if you don't specify.

Reply:
Ok so I've made gzip and uncompressed defaults to the write-csv benchmark. I also haven't added in the rest of them in right now as I'm not really sure there is a need (happy to be corrected there). cc @jonkeane

Reply:
Yup, agreed. The other compressions that are in arrow(bench) are only applicable to parquet or feather — gzip is the only compression we support for csv IIRC.